x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/aggs/InferencePipelineAggregationBuilder.java

```diff
@@ -199,6 +211,37 @@ public InferencePipelineAggregationBuilder rewrite(QueryRewriteContext context)
                     delegate.onFailure(LicenseUtils.newComplianceException(XPackField.MACHINE_LEARNING));
                 }
             }));
+
+
+        context.registerAsyncAction((client, listener) -> {
+            if (licenseState.isSecurityEnabled()) {
+                // check the user has ml privileges
+                SecurityContext securityContext =new SecurityContext(Settings.EMPTY, client.threadPool().getThreadContext());
+                useSecondaryAuthIfAvailable(securityContext, () -> {
+                    final String username = securityContext.getUser().principal();
+                    final HasPrivilegesRequest privRequest = new HasPrivilegesRequest();
+                    privRequest.username(username);
+                    privRequest.clusterPrivileges(GetTrainedModelsAction.NAME);
+                    privRequest.indexPrivileges(new RoleDescriptor.IndicesPrivileges[]{});
```

Comment:
Do we need to set index and application privileges even though they're empty?

Reply:
Yes because of this validation

https://github.com/elastic/elasticsearch/blob/master/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/user/HasPrivilegesRequest.java#L46